### PR TITLE
Save error status to request

### PR DESCRIPTION
**Pull request by Benzbeeb**

### CODE REVIEW

File: chain/x/zoracle/internal/types/request.go [hunk 1]
```diff
@@ -4,6 +4,14 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+type ResolveStatus int
+
+const (
+	Pending ResolveStatus = iota
```
Comment (sorawit) [inline chain/x/zoracle/internal/types/request.go]: Let's call them `Open`, `Success`, `Failure`. Pending is a bit confusing with pending list.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/handler.go [hunk 2]
```diff
@@ -150,15 +150,17 @@ func handleEndBlock(ctx sdk.Context, keeper Keeper) sdk.Result {
 
 		// TODO: Handle error if happen
 		if errOwasm != nil {
+			keeper.SetResolve(ctx, requestID, types.Failure)
```
Comment (taobun) [inline chain/x/zoracle/handler.go:152]: I think we can remove TODO

File: chain/x/zoracle/internal/keeper/report.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ func (k Keeper) AddReport(
 		return err
 	}
 
-	if request.IsResolved || request.ExpirationHeight < ctx.BlockHeight() {
+	if request.ResolveStatus == types.Success || request.ExpirationHeight < ctx.BlockHeight() {
```
Comment (taobun) [inline chain/x/zoracle/internal/keeper/report.go]: It would be `!= types.Open` ?

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]
looks great!

**Review by taobun — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Pull request merged**